src/backend/base/langflow/components/prototypes/UpdateData.py

```diff
@@ -1,40 +1,119 @@
+            default_keys = ["code", "_type", "number_of_fields", "text_key", "old_data", "text_key_validator"]
+            try:
+                field_value_int = int(field_value)
+            except ValueError:
+                return build_config
+            existing_fields = {}
+            if field_value_int > 15:
+                build_config["number_of_fields"]["value"] = 15
+                raise ValueError("Number of fields cannot exceed 15. Try using a Component to combine two Data.")
+            if len(build_config) > len(default_keys):
+                # back up the existing template fields
+                for key in build_config.copy():
+                    if key not in default_keys:
+                        existing_fields[key] = build_config.pop(key)
+
+            for i in range(1, field_value_int + 1):
+                key = f"field_{i}_key"
+                if key in existing_fields:
+                    field = existing_fields[key]
+                    build_config[key] = field
+                else:
+                    field = DictInput(
+                        display_name=f"Field {i}",
+                        name=key,
+                        info=f"Key for field {i}.",
+                        input_types=["Text", "Data"],
+                    )
+                    build_config[field.name] = field.to_dict()
+
+            build_config["number_of_fields"]["value"] = field_value_int
+        return build_config
+
+    async def build_data(self) -> Data:
+        new_data = self.get_data()
+        self.old_data.data.update(new_data)
+        if self.text_key:
+            self.old_data.text_key = self.text_key
+        self.status = self.old_data
+        self.add_validator(self.old_data)
+        return self.old_data
+
+    def get_data(self):
+        """Function to get the Data from the attributes"""
+        data = {}
+        for value_dict in self._attributes.values():
+            if isinstance(value_dict, dict):
+                # Check if the value of the value_dict is a Data
+                value_dict = {
+                    key: value.get_text() if isinstance(value, Data) else value for key, value in value_dict.items()
+                }
+                data.update(value_dict)
         return data
+
+    def add_validator(self, data: Data):
+        """This function validates that the Text Key is one of the keys in the Data"""
+        data_keys = data.data.keys()
+        if self.text_key not in data_keys and self.text_key != "":
+            raise ValueError(f"Text Key: {self.text_key} not found in the Data keys: {','.join(data_keys)}")
+
+    def post_code_processing(self, new_frontend_node: dict, current_frontend_node: dict):
+        """
+        This function is called after the code validation is done.
+        """
+        frontend_node = super().post_code_processing(new_frontend_node, current_frontend_node)
+        frontend_node["template"] = self.update_build_config(
```

Comment:
hm - I actually thought the `update_build_config` endpoint was already invoked when a field was changed. When is code validation done?

Reply:
Not sure but I believe the `update_build_config ` config is invoked in side the `post_code_processing` 
Also `post_code_processing` is a function of  class CustomComponent . We are overriding it  with the updated build_configs as soon as the number of fields is changed by the user in ui.